### PR TITLE
Automation templates [MAILPOET-4745]

### PR DESCRIPTION
**Pull request by websupporter**
## Description

Adds the templates and template stubs

## Linked PRs
https://github.com/mailpoet/mailpoet-premium/pull/650

## Linked tickets

[MAILPOET-4745]

[MAILPOET-4745]: https://mailpoet.atlassian.net/browse/MAILPOET-4745?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php [hunk 1]
```diff
@@ -53,22 +53,138 @@ function(WorkflowTemplate $template) use ($category): bool {
...
 
   private function createTemplates(): array {
-    $simpleWelcomeEmail = new WorkflowTemplate(
-      'simple-welcome-email',
+    $subscriberWelcomeEmail = new WorkflowTemplate(
+      'subscriber-welcome-email',
       WorkflowTemplate::CATEGORY_WELCOME,
-      "Automation template description is going to be here. Let's describe a lot of interesting ideas which incorporated into this beautiful and useful template",
+      __(
+        "Send a welcome email when someone subscribes to your list. Optionally, you can choose to delay this email by a couple of hours or days.",
```
Comment (JanJakes) [inline mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php]: The `by a couple of hours or days ` for all emails that have it... we also support minutes now. Is there a better term to describe this? @NeosinneR

File: mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx [hunk 2]
```diff
@@ -45,19 +51,50 @@ export function TemplateListItem({
...
+            setShowPremium(true);
+            return;
+          }
           void createWorkflowFromTemplate();
         }}
       >
+        <div className="badge">
+          {template.type === 'coming-soon' && (
+            <span>{__('Coming soon', 'mailpoet')}</span>
+          )}
```
Comment (JanJakes) [inline mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx:75]: It confused me a bit that I couldn't click on some emails. Then I realized those were not `Premium` but `Coming soon`. I think in Figma, the `Coming soon` category is grayed out a bit:

Screenshot 2022-11-08 at 15 50 39

Adding disabled cursor may help as well.

Shouldn't the `Coming soon` badge also have a different color than `Premium`, @NeosinneR?

File: mailpoet/assets/css/src/mailpoet-automation-templates.scss [hunk 3]
```diff
@@ -17,17 +17,25 @@ ul.mailpoet-automation-templates {
...
+    align-content: baseline;
+    align-items: flex-start;
     background: #fff;
     border: 1px solid #dcdcde;
     border-radius: 4px;
     cursor: pointer;
-    display: flex;
-    flex-direction: column;
+    display: grid;
+    grid-template-rows: 48px auto auto;
```
Comment (JanJakes) [inline mailpoet/assets/css/src/mailpoet-automation-templates.scss]: I think something like `40px` looks closer to Figma. The label seems to be 15px high, the space between the heading and the label is 24px in Figma, so we may go even slightly below 40px, actually.

**Review by JanJakes — COMMENTED** [review]
Looks nice and works!

Some tests seem to be failing, probably not flakiness.

I also noticed there's a "merge trunk" commit. Can we rebase and drop it somehow?

I was also thinking we may prefix the template slugs with the integration key, sth like `mailpoet:subscriber-welcome-email` and rename `slug` to `key`, but we can do such a polishing separately.

File: mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php (hunk 1 above)
Comment (NeosinneR) [reply mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php]: How about `Optionally, you can choose to send this email after a specified period.`?

File: mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx (hunk 2 above)
Comment (NeosinneR) [reply mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx:75]: The coming soon should indeed be grayed out. A disabled cursor can also be added.

As for the premium badge - yeah, can we try the MailPoet orange color (with white font) if it's better?

File: mailpoet/assets/css/src/mailpoet-automation-templates.scss (hunk 3 above)
Comment (websupporter) [reply mailpoet/assets/css/src/mailpoet-automation-templates.scss]: 40px in [13a7557](https://github.com/mailpoet/mailpoet/pull/4506/commits/13a75578d6684a1bddb9a13e2c8ccbd417f8461a)

File: mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx (hunk 2 above)
Comment (websupporter) [reply mailpoet/assets/js/src/automation/templates/components/template-list-item.tsx:75]: [6323825](https://github.com/mailpoet/mailpoet/pull/4506/commits/6323825bb514b774cccfea3495e8c5b6c468cbf2) and [3507ce7](https://github.com/mailpoet/mailpoet/pull/4506/commits/3507ce77fbe9880e2c46cf0c428ef013723888cf)

File: mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php (hunk 1 above)
Comment (websupporter) [reply mailpoet/lib/Automation/Engine/Storage/WorkflowTemplateStorage.php]: [757d1e0](https://github.com/mailpoet/mailpoet/pull/4506/commits/757d1e05b8382f27dc12bcfe3910bd7f3b09e2f9)

**Comment by websupporter** [comment]
@JanJakes thanks for the review. Rebased and got rid of the merge commit. Good catch.

**Review by JanJakes — APPROVED** [review]
Nice!

**Review by JanJakes — APPROVED** [review]

**Pull request merged**